### PR TITLE
Fix some Javadoc issues

### PR DESCRIPTION
**Pull request by offa**
Fixes some minor Javadoc issues.

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by jtnord — CHANGES_REQUESTED** [review]
some of the introductions are incorrect, esp in the definition list.
on top of this the build now produces 3 new warnings which where not present before the changes.

```
[INFO] --- javadoc:3.5.0:javadoc (default-cli) @ ldap ---
[INFO] Configuration changed, re-generating javadoc.
[WARNING] Javadoc Warnings
[WARNING] C:\workarea\source\github\jenkinsci\ldap-plugin\src\main\java\hudson\security\LDAPSecurityRealm.java:117: warning: empty <p> tag
[WARNING] * <p>
[WARNING] ^
[WARNING] C:\workarea\source\github\jenkinsci\ldap-plugin\src\main\java\hudson\security\LDAPSecurityRealm.java:229: warning: empty <p> tag
[WARNING] * <p>
[WARNING] ^
[WARNING] C:\workarea\source\github\jenkinsci\ldap-plugin\src\main\java\hudson\security\LDAPSecurityRealm.java:233: warning: empty <p> tag
[WARNING] * <p>
[WARNING] ^
[WARNING] 3 warnings
[INFO] ------------------------------------------------------------------------
[
```

File: src/main/java/hudson/security/LDAPSecurityRealm.java [hunk 1]
```diff
@@ -226,11 +226,11 @@
  * Because this is a standard, we expect most LDAP servers out there to use it, although
  * there are different objectClasses that can be used for similar purposes, and apparently
  * many deployments choose to use different objectClasses.
- *
+ * <p>
```
[the remark this answers is not in the document]
Comment (offa) [reply src/main/java/hudson/security/LDAPSecurityRealm.java]: :man_facepalming: Thanks, updated.

File: src/main/java/hudson/security/LDAPSecurityRealm.java [hunk 2]
```diff
@@ -113,7 +113,7 @@
 
 /**
  * {@link SecurityRealm} implementation that uses LDAP for authentication.
- *
+ * <p>
```
Comment (jtnord) [inline src/main/java/hudson/security/LDAPSecurityRealm.java:116]:
```suggestion
```

**Comment by offa** [comment]
What's left to do here @jtnord

File: src/main/java/hudson/security/LDAPSecurityRealm.java [hunk 3]
```diff
@@ -254,7 +251,7 @@ public class LDAPSecurityRealm extends AbstractPasswordBasedSecurityRealm {
 
     /**
      * The root DN to connect to. Normally something like "dc=sun,dc=com"
-     *
+     * <p>
```
Comment (olamy) [inline src/main/java/hudson/security/LDAPSecurityRealm.java:254]: tbh honest, even if non failing. I don't like those non closed elements :) 
maybe we should have the checsktyle rule "Tight-HTML" to enforce that.

File: src/main/java/hudson/security/LDAPSecurityRealm.java [hunk 4]
```diff
@@ -284,7 +281,7 @@ public class LDAPSecurityRealm extends AbstractPasswordBasedSecurityRealm {
 
     /**
      * Query to locate an entry that identifies the user, given the user name string.
-     *
+     * <p>
```
Comment (jtnord) [inline src/main/java/hudson/security/LDAPSecurityRealm.java:284]: this change and many like it do not seem correct to me - `Normally "uid={0}"` is not a paragraph.  It is strongly related to the previous sentence and to me does not seem to make grammatical sense to me.
I have not requested changes or approved as I was ambivalent on this, and as such decided to let the reset of the maintainers make a decision.

There is no generally need AFAICT to separate the introductory sentence from any following ones.  they often can and do make sense together.

File: src/main/java/hudson/security/LDAPSecurityRealm.java (hunk 3 above)
Comment (offa) [reply src/main/java/hudson/security/LDAPSecurityRealm.java:254]: I don't have a preference on this, we can use `<p>…</p>` too.

File: src/main/java/hudson/security/LDAPSecurityRealm.java (hunk 4 above)
Comment (offa) [reply src/main/java/hudson/security/LDAPSecurityRealm.java:284]: A p-element for separating examples or snippets is not uncommon, e.g. in Javadoc of Java classes it is occasionally seen.

**Pull request closed without merging**